lib/vquic/msh3.c

```diff
@@ -110,19 +119,22 @@ static CURLcode msh3_data_setup(struct Curl_cfilter *cf,
   struct HTTP *stream = data->req.p.http;
   (void)cf;
 
-  H3BUGF(infof(data, "msh3_data_setup"));
-  stream->recv_buf = malloc(MSH3_REQ_INIT_BUF_LEN);
+  DEBUGASSERT(stream);
   if(!stream->recv_buf) {
```

Comment:
Why would it not already be `NULL`?

Reply:
There is another path via `DATA_SETUP` event that might already initialize this.

Reply:
Could you explain the state machine a little?